src/analysis/processing/qgsalgorithmexportgeometryattributes.cpp

```diff
@@ -0,0 +1,273 @@
+    begin                : February 2025
+    copyright            : (C) 2025 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmexportgeometryattributes.h"
+#include "qgsunittypes.h"
+#include "qgsgeometrycollection.h"
+#include "qgscurve.h"
+
+///@cond PRIVATE
+
+QString QgsExportGeometryAttributesAlgorithm::name() const
+{
+  return QStringLiteral( "exportaddgeometrycolumns" );
+}
+
+QString QgsExportGeometryAttributesAlgorithm::displayName() const
+{
+  return QObject::tr( "Add geometry attributes" );
+}
+
+QStringList QgsExportGeometryAttributesAlgorithm::tags() const
+{
+  return QObject::tr( "export,add,information,measurements,areas,lengths,perimeters,latitudes,longitudes,x,y,z,extract,points,lines,polygons,sinuosity,fields" ).split( ',' );
+}
+
+QString QgsExportGeometryAttributesAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsExportGeometryAttributesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsExportGeometryAttributesAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Computes geometric properties of the features in a vector layer. Algorithm generates a new "
+                      "vector layer with the same content as the input one, but with additional attributes in its "
+                      "attributes table, containing geometric measurements.\n\n"
+                      "Depending on the geometry type of the vector layer, the attributes added to the table will "
+                      "be different." );
+}
+
+QgsExportGeometryAttributesAlgorithm *QgsExportGeometryAttributesAlgorithm::createInstance() const
+{
+  return new QgsExportGeometryAttributesAlgorithm();
+}
+
+void QgsExportGeometryAttributesAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ), QList<int>() << static_cast<int>( Qgis::ProcessingSourceType::VectorAnyGeometry ) ) );
+
+  const QStringList options = QStringList()
+                              << QObject::tr( "Layer CRS" )
+                              << QObject::tr( "Project CRS" )
+                              << QObject::tr( "Ellipsoidal" );
```

Comment:
Let's be super explicit here -- I think there's a lot of room for confusion here:

```suggestion
                              << QObject::tr( "Cartesian Calculations in Layer's CRS" )
                              << QObject::tr( "Cartesian Calculations in Project's CRS" )
                              << QObject::tr( "Ellipsoidal Calculations" );
```